cogito/protocol.go

```diff
@@ -245,7 +253,18 @@ func (src *Source) Validate() error {
 		if src.Repo == "" {
 			mandatory = append(mandatory, "repo")
 		}
-		if src.AccessToken == "" {
+
+		if src.GitHubApp != (github.GitHubApp{}) {
```

Comment:
also not a fan of:

```
if ....
   if ...
else if ...
```
I tried many things... This is the cleanest so far... Any suggestion to improve?

Reply:
I think that the current code is fine.

Note though I think a bug:
```
if src.GitHubApp != (github.GitHubApp{}) {
...
} else if src.AccessToken == "" {
...
}
```

it seems that src.AccessToken and src.GitHubApp  are mutually exclusive, but the code does not enforce this.

Reply:
I mean: I think we need something like (pseudocode):

if src.GitHubApp != (github.GitHubApp{}) && src.AccessToken != "" {
    error: cannot specify both
}

or something similar. But now I am even more confused, since we have
```
else if src.AccessToken == "" 
```
while I was expecting
```
else if src.AccessToken != "" 
```

Maybe I got it wrong...

Reply:
Fixed...